CRM/Event/Form/Search.php

```diff
@@ -447,7 +447,7 @@ public function setDefaultValues() {
       elseif (is_array($status) && !array_key_exists('IN', $status)) {
         $statusTypes = array_keys($status);
       }
-      $this->_defaults['participant_status_id'] = is_array($statusTypes) ? array_keys($statusTypes) : $statusTypes;
+      $this->_defaults['participant_status_id'] = $statusTypes;
```

Comment:
Up on lines 439 and 442 you'll need to call array_keys on the value, otherwise they don't work anymore.

Reply:
Hi @pradpnayak this just needs a tweak so it handles array_keys consistently for all sections in the `if` statement. Either adding it in the if block or putting back the array_keys here and taking it out of the if block.